File: server/routes/reviewRoutes.js
const express = require('express');
const router = express.Router();
const reviewController = require('../controllers/reviewController');
const authMiddleware = require('../middleware/authMiddleware');

router.get('/:vendorId', reviewController.getVendorReviews);
router.post('/', (req, res, next) => {
  // Optional auth
  if (req.headers.authorization) {
    return authMiddleware(req, res, () => reviewController.createReview(req, res, next));
  }
  reviewController.createReview(req, res, next);
});

module.exports = router;

File: server/routes/review_routes.py
from flask import Blueprint
from controllers.review_controller import get_vendor_reviews, create_review
from middleware.auth_middleware import jwt_required_custom

review_bp = Blueprint('review_bp', __name__)

review_bp.route('/vendor/<int:vendor_id>', methods=['GET'])(get_vendor_reviews)
review_bp.route('', methods=['POST'])(jwt_required_custom(create_review))
